ensime_shared/client.py

```diff
@@ -219,14 +243,16 @@ def disable_completely(e):
                 port = self.ensime.http_port()
                 uri = "websocket" if server_v2 else "jerky"
                 self.ensime_server = gconfig["ensime_server"].format(port, uri)
-            with catch(websocket.WebSocketException, disable_completely):
+            with catch(Exception, disable_completely):
                 # Use the default timeout (no timeout).
-                options = {"subprotocols": ["jerky"]} if server_v2 else {}
-                options['enable_multithread'] = True
-                self.log.debug("About to connect to %s with options %s",
-                               self.ensime_server, options)
-                self.ws = websocket.create_connection(self.ensime_server, **options)
-            if self.ws:
+                async def connect():
+                    options = {"subprotocols": ["jerky"]} if server_v2 else {}
+                    self.log.debug("About to connect to %s with options %s",
+                                self.ensime_server, options)
+                    self.ws = await websockets.client.connect(self.ensime_server, **options)
+                    True
+                gotws = asyncio.run_coroutine_threadsafe(connect(), self.ws_loop).result(10)
+            if gotws:
```

Comment:
is this `gotws` needed at all? if future fails it will throw exception, so should be ok w/o it and the following line shifted right into the catch block